File: frontend/src/app/dto/table-data-dto.ts
import { SortDirection } from "@angular/material/sort";

export class TableDataDto {
    private _page: string;
    private _pageSize: string;
    private _sortHeader: string;
    private _direction: SortDirection;
    private _globalFilter: string;

    constructor() {
        this._page = '0';
        this._pageSize = '10';
        this._sortHeader = 'id';
        this._direction = 'asc';
        this._globalFilter = '';
    }

    public get page(): string {
        return this._page;
    }

    public set page(value: string) {
        this._page = value;
    }

    public get pageSize(): string {
        return this._pageSize;
    }

    public set pageSize(value: string) {
        this._pageSize = value;
    }

    public get sortHeader(): string {
        return this._sortHeader;
    }

    public set sortHeader(value: string) {
        this._sortHeader = value;
        if (value == null) {
            this._sortHeader = ''
        }
    }

    public get direction(): SortDirection {
        return this._direction;
    }

    public set direction(value: SortDirection) {
        this._direction = value;
    }

    public get globalFilter(): string {
        return this._globalFilter;
    }

    public set globalFilter(value: string) {
        this._globalFilter = value;
    }
}

File: backend/src/main/java/pl/kamil/backend/dto/TableDataDto.java
package pl.kamil.backend.dto;

import java.util.Optional;

import org.springframework.data.domain.Sort.Direction;

public class TableDataDto {
    private int _page;
    private int _pageSize;
    private String _sortHeader;
    private String _direction;
    private String _globalFilter;


    public TableDataDto() {
        _page = 0;
        _pageSize = Integer.MAX_VALUE;
        _sortHeader = "id";
        _direction = "ASC";
        _globalFilter = "";
    }

    public int get_page() {
        return _page;
    }

    public void set_page(int page) {
        this._page = page;
    }

    public int get_pageSize() {
        return _pageSize;
    }

    public void set_pageSize(int limit) {
        this._pageSize = limit;
    }

    public String get_sortHeader() {
        return _sortHeader;
    }

    public void set_sortHeader(String sort) {
        this._sortHeader = sort;
        if (sort.isEmpty())
            this._sortHeader = "id";
    }

    public String get_direction() {
        return _direction;
    }

    public void set_direction(String order) {
        Optional<Direction> direction = Direction.fromOptionalString(order);
        if(!direction.isPresent()){
            this._direction = Direction.ASC.name();
            return;
        }
        this._direction = direction.get().name();
    }

    public String get_globalFilter() {
        return _globalFilter;
    }

    public void set_globalFilter(String globalFilter) {
        this._globalFilter = globalFilter;
    }

}
